src/Discord.Net.Rest/Entities/Users/UserHelper.cs

```diff
@@ -84,5 +88,25 @@ public static async Task RemoveRolesAsync(IGuildUser user, BaseDiscordClient cli
             foreach (var roleId in roleIds)
                 await client.ApiClient.RemoveRoleAsync(user.Guild.Id, user.Id, roleId, options).ConfigureAwait(false);
         }
+
+        public static async Task SetTimeoutAsync(IGuildUser user, BaseDiscordClient client, TimeSpan span, RequestOptions options)
+        {
+            if (span.TotalDays >= 28)
+                throw new ArgumentOutOfRangeException(nameof(span), "Offset cannot be more than 28 days from the current date.");
```

Comment:
Just out of curiosity: Why is this feature limited? Is it a mandatory limit by Discord API?

Reply:
Yeah it's a limit imposed by discord as per https://discord.com/developers/docs/resources/guild#modify-guild-member

![image](https://user-images.githubusercontent.com/25526843/146990722-5c3496ca-a5e9-460f-8dce-589ea83f265d.png)

Reply:
There should be another check for negative timespans